src/fsharp/infos.fs

```diff
@@ -1005,7 +1003,8 @@ type MethInfo =
     member x.FormalMethodTypars = 
         match x with 
         | ILMeth(_,ilmeth,_) -> ilmeth.FormalMethodTypars
-        | FSMeth(g,typ,vref,_) ->  
+        | FSMeth(g,_,vref,_) ->  
+            let typ = x.ApparentEnclosingAppType
```

Comment:
Ugh, yes. However there are a few other paths to `AnalyzeTypeOfMemberVal`, which is assuming that `typ` is an AppTy, e.g. via `GetCompiledReturnTy` and `GetParamTypes` which I think will also fail.  I'm surprised we're not hitting those

Reply:
@dsyme, yes I was going to look through those today.  The trick is figuring out the repro, as you say, with all this code base none hit it.